RocketChat/Rocket.Chat.ReactNative
# [NEW] Add recording, uploading activity indicators

Setup [#222392](https://github.com/RocketChat/Rocket.Chat/pull/22392) on server.

### Recording, Uploading activities
https://user-images.githubusercontent.com/47038980/142961215-62f4258f-28bb-457b-b6c7-2ebba9f036a7.mov

#### Thread activities
https://user-images.githubusercontent.com/47038980/142961189-38af257d-13d8-4f7d-9b9c-a9de65b9e875.mov



## Proposed changes


## Issue(s)	


## How to test or reproduce


First of all, you need to check which version type of the server you are on. If you are in the servers with a version equal to or greater than 4.0.0 the indicators will appear as demonstrated in the video, and will happen for rooms and threads. Adding to this, the test plan is:
- You should be logged with 2 different accounts into different devices.
- Then go to the same room, with both users
- And when you are typing/recording/uploading with one account, should appear the same indicator to the other user
- Besides, if you are uploading a bigger file and in parallel is typing, the other user should read that: `you are uploading, you are typing`
- And vice versa too


Although, if you are on a server with a version less than 4.0.0 the indicators should occur just into rooms and only for typing

## Screenshots

## Types of changes




## Checklist



## Further comments


# web app integration 
[#222327](https://github.com/RocketChat/Rocket.Chat/pull/22392)
